tools/c7n_azure/c7n_azure/resources/vm.py

```diff
@@ -55,36 +55,52 @@ class VmStopAction(BaseAction):
 
     schema = type_schema('stop')
 
+    def stop(self, resource_group, vm_name):
+        self.client.virtual_machines.power_off(resource_group, vm_name)
+
     def process(self, vms):
+        self.client = self.manager.get_client()
         for vm in vms:
-            self.manager.get_client().virtual_machines.power_off(vm['resourceGroup'], vm['name'])
+            self.stop(vm['resourceGroup'], vm['name'])
 
 
 @VirtualMachine.action_registry.register('start')
 class VmStartAction(BaseAction):
 
     schema = type_schema('start')
 
+    def start(self, resource_group, vm_name):
+        self.client.virtual_machines.start(resource_group, vm_name)
+
     def process(self, vms):
+        self.client = self.manager.get_client()
         for vm in vms:
-            self.manager.get_client().virtual_machines.start(vm['resourceGroup'], vm['name'])
+            self.start(vm['resourceGroup'], vm['name'])
 
 
 @VirtualMachine.action_registry.register('restart')
 class VmRestartAction(BaseAction):
 
     schema = type_schema('restart')
 
+    def restart(self, resource_group, vm_name):
+        self.client.virtual_machines.restart(resource_group, vm_name)
+
     def process(self, vms):
+        self.client = self.manager.get_client()
```

Comment:
Functionally this is exactly right but in terms of Python convention, I believe it is considered inappropriate to set class instance variables outside of of `__init__` because it makes it harder to keep track of them.  You can just move this to `__init__` and  it should work the same.

For example here is the init from `class Tag(BaseAction):`

```
    def __init__(self, data=None, manager=None, log_dir=None):
        super(Tag, self).__init__(data, manager, log_dir)
        self.client = self.manager.get_client('azure.mgmt.resource.ResourceManagementClient')
```